tunnel/intra/doh/doh.go

```diff
@@ -129,19 +131,23 @@ func (t *transport) dial(network, addr string) (net.Conn, error) {
 			// Don't try this IP twice.
 			continue
 		}
-		if conn, err = split.DialWithSplitRetry(tcpaddr(ip), tcpTimeout, nil); err == nil {
+		if conn, err = split.DialWithSplitRetry(t.dialer, tcpaddr(ip), nil); err == nil {
 			log.Infof("Found working IP: %s", ip.String())
 			return conn, nil
 		}
 	}
 	return nil, err
 }
 
-// NewDoHTransport returns a DoH DNSTransport, ready for use.
+// NewTransport returns a DoH DNSTransport, ready for use.
 // This is a POST-only DoH implementation, so the DoH template should be a URL.
-// addrs is a list of domains or IP addresses to use as fallback, if the hostname
-// lookup fails or returns non-working addresses.
-func NewTransport(rawurl string, addrs []string, listener Listener) (Transport, error) {
+// `rawurl` is the DoH template in string form.
+// `addrs` is a list of domains or IP addresses to use as fallback, if the hostname
+//   lookup fails or returns non-working addresses.
+// `protector` is the socket protector to apply to all outbound sockets, to ensure
+//   they aren't routed back into the VPN.
+// `listener` will receive the status of each DNS query when it is complete.
+func NewTransport(rawurl string, addrs []string, protector protect.Protector, listener Listener) (Transport, error) {
```

Comment:
Pass a dialer instead of a protector.

This code doesn't need to know about protection of sockets, a VPN thing. It just needs a way to create sockets.

Reply:
Done!